lib/matplotlib/axes/_base.py

```diff
@@ -985,14 +1040,30 @@ def clear(self):
         """clear the axes"""
         self.cla()
 
+    def set_prop_cycle(self, prop_cycle):
+        """
+        Set the prop cycle for any future plot commands on this Axes.
+
+        *prop_cycle* is a :class:Cycler object.
+        Can also be `None` to reset to the cycle defined by the
+        current style.
+        """
+        self._get_lines.set_prop_cycle(prop_cycle)
+        self._get_patches_for_fill.set_prop_cycle(prop_cycle)
+
     def set_color_cycle(self, clist):
         """
         Set the color cycle for any future plot commands on this Axes.
 
         *clist* is a list of mpl color specifiers.
+
+        .. deprecated:: 1.5
```

Comment:
Replacing the current cycle with one that just does color replicates the current behavior so any users using this will not have to change anything.

Someplace we talked about updating cyclers in place and decided against it due to the complication involved.  `Cycler` does not currently expose enough detail to introspect and tell the difference between `cycler('a', 'aa') + cycler('b', 'bb')` and `cycler('a', 'a') * cycler('b', 'bb')` which I think is a feature, not a bug.  It is also not clear what will happen if you update the underlying data while you are holding an active iterator over the cycler.

It is better to do the simple easy to understand thing that will always work than the possibly more clever, but very brittle thing.